ai/executors/pathfinder_module.py

```diff
@@ -33,21 +34,29 @@ def exec(self, game_state, ai_cmds) -> Dict:
 
         return self.paths
 
-    def updates_obstacles(self, game_state):
+    def updates_obstacles(self, game_state: GameState):
+
         self.obstacles.clear()
 
         our_team = [player for player in game_state.our_team.available_players.values()]
         enemy_team = [player for player in game_state.enemy_team.available_players.values()]
 
         for other in our_team + enemy_team:
-            self.obstacles.append(other.pose.position)
+            self.obstacles.append(Obstacle(other.position.array, avoid_distance=MIN_DISTANCE_FROM_OBSTACLE))
+
 
-    def player_obstacles(self, game_state, player, ai_cmd):
+    def player_optionnal_obstacles(self, game_state: GameState, ai_cmd: AICommand) -> List[Obstacle]:
         path_obstacles = self.obstacles.copy()
 
         if ai_cmd.ball_collision and game_state.is_ball_on_field:
-            path_obstacles.append(game_state.ball_position)
+            path_obstacles.append(Obstacle(game_state.ball_position.array, avoid_distance=MIN_DISTANCE_FROM_OBSTACLE))
+
+        if ai_cmd.ball_collision and game_state.is_ball_on_field:
+            path_obstacles.append(Obstacle(game_state.ball_position.array, avoid_distance=MIN_DISTANCE_FROM_OBSTACLE))
 
-        path_obstacles.remove(player.pose.position)
+        path_obstacles.append(Obstacle(Position(-4500, 500).array, avoid_distance=1000))
```

Comment:
It would be preferable to use field constantes
```python
self.game_state.field.our_goal # or self.game_state.field.goal_area ..
```

Reply:
ca va etre dealer avec quand @sim0504 va merge son travail sur les objets de collisions. Ce détail la n'a pas vraiment a voir avec ma PR

Reply:
Mais si on merge cette pr sur dev, il va y avoir des terrains où il y aura des obstacles au mauvais endroit. Au pire, si cela n'a pas rapport à ta PR, on pourrait l'enlever et attendre la PR de Simon.
Qu'est-ce que tu en penses?

Reply:
jhfais le changement

Reply:
C FAIT ACCEPT LA PR N00B